libs/@local/codegen/src/definitions/primitive.rs

```diff
@@ -0,0 +1,32 @@
+use specta::datatype;
+
+#[derive(Debug, Clone)]
+pub enum Primitive {
+    String,
+    Number,
```

Comment:
I appreciate that for now the type here is the same, but would it make sense to (in the future) to refine this to e.g. `Integer` and `Number` (with signed and width) in particular as validators like zod, arktype or effect allow to enforce this from the typescript side and we're loosing quite some information here.

Additionally anything over the width of 53 (for integers) should become a BigInt, as anything above that turns into a lossy float.

Reply:
This can be changed if we actually construct these types in TypeScript. But for now, `number` is the only type we have in TS. See H-3988